api-challenge/src/main/java/nix/nhrin/apichallenge/service/DeIdentificationPatientRecordService.java

```diff
@@ -0,0 +1,75 @@
+package nix.nhrin.apichallenge.service;
+
+import com.opencsv.CSVReader;
+import lombok.SneakyThrows;
+import nix.nhrin.apichallenge.entity.Patient;
+import org.json.simple.JSONObject;
+import org.springframework.stereotype.Service;
+
+import java.io.FileReader;
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Service
+public class DeIdentificationPatientRecordService {
+
+    private final String EMAIL_REGEX = "[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+";
+    private final String US_SOCIAL_SECURITY_NUMBER_REGEX = "\\d{3}-?\\d{2}-?\\d{4}";
+    private final String US_TELEPHONE_NUMBER_REGEX = "(\\+\\d{1,3}( )?)?((\\(\\d{3}\\))|\\d{3})[- .]?\\d{3}[- .]?\\d{4}";
+    private final Map<String, Integer> DATE_REGEX = new HashMap<>() {{
+        put("\\s+(?:19\\d{2}|20\\d{2})[-/.](?:0[1-9]|1[012])[-/.](?:0[1-9]|[12][0-9]|3[01])\\b", 1); //YYYY/MM/DD
+        put("\\s+(?:0[1-9]|[12][0-9]|3[01])[-/.](?:0[1-9]|1[012])[-/.](?:19\\d{2}|20\\d{2})\\b", 7); //DD/MM/YYYY
+        put("\\s+(?:0[1-9]|1[012])[-/.](?:0[1-9]|[12][0-9]|3[01])[-/.](?:19\\d{2}|20\\d{2})\\b", 7); //MM/DD/YYYY
+    }};
+
+
+    public JSONObject deIdentificationAllInfo(Patient patient) {
```

Comment:
It's better to start method names with a verb. Like getDeiden...,  prepareDeiden..

Reply:
fixed